src/lib/getVisibilityWatcher.ts

```diff
@@ -18,6 +18,7 @@ import {onBFCacheRestore} from './bfcache.js';
 import {getActivationStart} from './getActivationStart.js';
 
 let firstHiddenTime = -1;
+const onHiddenFunctions: (() => void)[] = [];
```

Comment:
Optional: if you made this a `Set` instead of an array, then it would prevent you from adding the same function twice (which matches the native `addEventListener()` behavior). I don't think that will matter in this library, but in general I like to use Sets for these types of things to to keep the behavior the same.